### PR TITLE
vim: Add patch to resolve CVE-2024-41957 & CVE-2024-41965.

### PR DESCRIPTION
**Pull request by Sumynwa**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 
Add patch to resolve CVE-2024-41957 & CVE-2024-41965

Reference
https://github.com/vim/vim/security/advisories/GHSA-f9cr-gv85-hcr4
https://github.com/vim/vim/security/advisories/GHSA-46pw-v7qw-xc2f

###### Change Log  
- Add patch files

###### Does this affect the toolchain?  
NO

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-41957
- https://nvd.nist.gov/vuln/detail/CVE-2024-41965

###### Test Methodology

- Pipeline build id:[ Buddy Build](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=618868&view=results)

### CODE REVIEW

**Review by mfrw — APPROVED** [review]
LGTM :)

File: SPECS/vim/vim.spec [hunk 1]
```diff
@@ -2,14 +2,18 @@
...
+Release:        3%{?dist}
 License:        Vim
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          Applications/Editors
 URL:            https://www.vim.org
 Source0:        https://github.com/%{name}/%{name}/archive/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
 Patch0:         CVE-2024-22667.patch
+Patch1:         CVE-2024-41957.patch
+Patch2:         fix_save_unnamed_buffer_correctly.patch
```
Comment (jslobodzian) [inline SPECS/vim/vim.spec:14]: Is there a CVE associated with Patch2?  Is that the correct name?

Comment (Sumynwa) [reply SPECS/vim/vim.spec:14]: @jslobodzian No this patch is not part of CVE, but a general fix upstream (https://github.com/vim/vim/pull/14398). The patch was added here for the CVE patch to apply cleanly.

Comment (jslobodzian) [reply SPECS/vim/vim.spec:14]: @Sumynwa can you combine them into one patch file then?

Comment (Sumynwa) [reply SPECS/vim/vim.spec:14]: @jslobodzian Sure I will.

Comment (jslobodzian) [reply SPECS/vim/vim.spec:14]: @Sumynwa just waiting on the patch to be combined.

Comment (Sumynwa) [reply SPECS/vim/vim.spec]: @jslobodzian Patches have been merged.

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**